pkg/controller/node/node_controller.go

```diff
@@ -116,17 +118,25 @@ func (n *NodeController) SyncHandler(key string) error {
 	if err != nil {
 		return fmt.Errorf("error listing cloud private ip config, err: %v", err)
 	}
-	// Filter out cloudPrivateIPConfigs assigned to node (key) and write the entry
-	// into same slice starting from index 0, finally chop off unwanted entries
-	// when passing it into GetNodeEgressIPConfiguration.
-	index := 0
+	// Filter cloudPrivateIPConfigs that are associated with this node.
+	// Include CPICs where spec.node OR status.node matches, to handle transitions
+	// during move operations and avoid counting CPIC IPs as "consumed capacity".
+	// We need both spec.node and status.node because during a move operation:
+	// - The IP might still be on the old node (status.node) while spec.node has changed
+	// - The IP might already be on the new node (spec.node) while status.node hasn't updated
+	// Build set of CPIC-managed IP addresses for this node
+	// Include CPICs where spec.node OR status.node matches to handle transitions
+	cpicIPs := sets.New[string]()
 	for _, cloudPrivateIPConfig := range cloudPrivateIPConfigs {
-		if isAssignedCloudPrivateIPConfigOnNode(cloudPrivateIPConfig, key) {
```

Comment:
you may remove isAssignedCloudPrivateIPConfigOnNode method which is no longer needed.